src/AsyncSessionMiddlewareFactory.php

```diff
@@ -2,32 +2,39 @@
 
 namespace Interop\Session\Middleware\Async;
 
-use Psr\Http\Message\{ServerRequestInterface as Request, ResponseInterface as Response};
-use Interop\Session\Manager\{SessionManagerInterface as SessionManager, Utils\DefaultManager\DefaultSessionManager as DefaultManager};
+use Psr\Http\Message\{
+    ServerRequestInterface as Request, ResponseInterface as Response
+};
+use Interop\Session\Manager\{
+    SessionManagerInterface as SessionManager, Utils\DefaultManager\DefaultSessionManager as DefaultManager
+};
 use Interop\Session\Configuration\SessionConfigurationInterface;
 
 use Interop\Session\SessionInterface;
 
-class AsyncSessionMiddlewareFactory {
+class AsyncSessionMiddlewareFactory
+{
 
-  public static function create(?SessionConfigurationInterface $configuration = null) {
-    if (!$configuration) {
-      $configuration = new \Interop\Session\Manager\Utils\DefaultManager\SessionConfiguration();
-    }
-    $manager = new DefaultManager($configuration);
-    return new AsyncSessionMiddleware($manager);
-  }
+    public function __invoke(\Interop\Container\ContainerInterface $container)
+    {
+        $manager = $container->get(SessionManager::class);
+        if ($container->has(SessionManager::class)) {
```

Comment:
You always enter this if statement (because you perform a get on the same service the line above and if the service does not exist, it throws an exception)